### PR TITLE
Add custom scheduling for backup jobs

### PR DESCRIPTION
**Pull request by benjaminjb**
**Checklist:**
 - [ ] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change, as applicable?
 - [ ] Have you tested your changes on all related environments with successful results, as applicable?
   - [ ] Have you added automated tests?

**Type of Changes:**
 - [ ] New feature
 - [ ] Bug fix
 - [ ] Documentation
 - [ ] Testing enhancement
 - [ ] Other

**What is the current behavior (link to any open issues here)?**

Backup jobs aren't schedulable/don't copy affinity rules from elsewhere in the spec

**What is the new behavior (if this is a feature change)?**
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

Job now has its own section for affinity, toleration

**Other Information**:
Issue [sc-11582]

### CODE REVIEW

**Comment by benjaminjb** [comment]
Issue raised in https://github.com/CrunchyData/postgres-operator/issues/2827

File: pkg/apis/postgres-operator.crunchydata.com/v1beta1/pgbackrest_types.go [hunk 1]
```diff
@@ -171,6 +171,17 @@ type BackupJobs struct {
 	// More info: https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/
 	// +optional
 	PriorityClassName *string `json:"priorityClassName,omitempty"`
+
+	// Scheduling constraints of the Dedicated repo host pod.
+	// Changing this value causes repo host to restart.
```
Comment (cbandy) [inline pkg/apis/postgres-operator.crunchydata.com/v1beta1/pgbackrest_types.go:176]: ❓ These new fields are for jobs, rather than repo host, right?

Comment (benjaminjb) [reply pkg/apis/postgres-operator.crunchydata.com/v1beta1/pgbackrest_types.go]: updated

File: pkg/apis/postgres-operator.crunchydata.com/v1beta1/pgbackrest_types.go [hunk 2]
```diff
@@ -171,6 +171,16 @@ type BackupJobs struct {
 	// More info: https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/
 	// +optional
 	PriorityClassName *string `json:"priorityClassName,omitempty"`
+
+	// Scheduling constraints of pgBackRest backup Job pods.
+	// More info: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node
+	// +optional
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+
+	// Tolerations of pgBackRest backup Job pods. Changing this value causes a restart.
```
Comment (cbandy) [inline pkg/apis/postgres-operator.crunchydata.com/v1beta1/pgbackrest_types.go:180]: ❓ What gets restarted? 🤔 Does the CronJob controller kill running Jobs when a CronJob spec changes?

(Maybe this comment is leftover from something else?)

Comment (benjaminjb) [reply pkg/apis/postgres-operator.crunchydata.com/v1beta1/pgbackrest_types.go:180]: Yeah, we ran into this when we were working on upgrades as well -- we have comments on a few fields about how "changes cause restarts", but I'm not sure where those comments originate or if they are accurate (across our range of supported kube envs).

Comment (benjaminjb) [reply pkg/apis/postgres-operator.crunchydata.com/v1beta1/pgbackrest_types.go:180]: (that said, the notion of a job pod getting restarted seems a little different than the notion of a longer-lived pod getting restarted)

Comment (cbandy) [reply pkg/apis/postgres-operator.crunchydata.com/v1beta1/pgbackrest_types.go:180]: I see we're not setting [`concurrencyPolicy`](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/cron-job-v1/#CronJobSpec) and I see that becomes `concurrencyPolicy: Allow` in Kubernetes 1.19.

🔧 I don't think this field will kill any jobs, so let's leave the comment out.

Comment (benjaminjb) [reply pkg/apis/postgres-operator.crunchydata.com/v1beta1/pgbackrest_types.go:180]: What do you think about `PriorityClassName` -- that also has a restart comment, but would that follow the same logic?

Comment (cbandy) [reply pkg/apis/postgres-operator.crunchydata.com/v1beta1/pgbackrest_types.go:180]: Yep, totally. These fields don't affect Postgres.

**Pull request merged**